full-stack-practice/client/src/lib/read.ts

```diff
@@ -0,0 +1,9 @@
+import { products, type Product } from './data';
+
+export function readCatalog(): Promise<Product[]> {
+  return Promise.resolve(products);
```

Comment:
This (and the function below) need to be changed to use `fetch` to call your backend API. Otherwise, this isn't a full stack application!